setup.py

```diff
@@ -12,10 +12,12 @@
     python_requires='>=3.8',
     install_requires=[
         "backoff",
+        "faiss-cpu",
```

Comment:
I suggest we remove `faiss-cpu` and `sentence-transformers`. These are both heavy and optional dependencies.

Instead, either the kNN notebook should instruct users to install faiss (link to https://github.com/facebookresearch/faiss/blob/main/INSTALL.md) and then sentence-transformers (pip).

OR, if these don't exist, maybe it makes sense to pip install them inside the code itself?

Basically, faiss-cpu will break repos that already have the conda version of GPU faiss. So we'd want to avoid that. Perhaps many users (who will have tiny training sets) will not need kNN right away, so it makes sense to keep the dependencies optional?

Reply:
> pip install them inside the code itself?

I don't think this is a good practice to install something without notice or asking or changing env in general. 
Just mentioning that a user needs to install smth in a notebook is ok. I'll add try-except catches to the code and notes to the notebook.